src/main.rs

```diff
@@ -13,7 +13,7 @@ use tracing_subscriber::{fmt, prelude::*, EnvFilter};
 /// all subsequent arguments up to "--". If a "--" argument is present, all
 /// arguments after it are executed as a command if any input is newer than all
 /// outputs.
-fn main() {
+fn main() -> Result<(), Error> {
```

Comment:
This seems to be frowned upon by Google, preferring exit(). I would imagine one downside is that you have no control over the formatting.

Reply:
- You can use `?` in `main`, the same as anywhere else
- The [`Termination` trait](https://doc.rust-lang.org/stable/std/process/trait.Termination.html) is stable and controls behaviour like formatting and exit codes

Use `fn main() -> ExitCode` and `return ExitCode::SUCCESS` / `return ExitCode::from(123)` / `return 123u8.into()`. IMHO calling _any_ function that doesn't return (exit, panic, …) is a smell.

Anyway, at least when I read it last year, my opinion of Google's Rust style guide was not high…